Implemented BookDao

svitlana-tertyshna opened the pull request:
Created a Book model, created interface Dao and BookDao implementation, created DB and ConnectionUtil, realized call of all CRUD methods in the Main class

svitlana-tertyshna committed:
Implemented BookDao

Created a Book model, created interface Dao and BookDao implementation, created DB and ConnectionUtil, realized call of all CRUD methods in the Main class

liliia-ponomarenko commented on src/main/java/mate/academy/lib/dao/Dao.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.List;
+import java.util.Optional;
+
+@mate.academy.lib.Dao
```
```suggestion
@Dao
```
add import

liliia-ponomarenko replied:
use this annotation only for implementation

liliia-ponomarenko commented on src/main/java/mate/academy/lib/dao/Dao.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.Optional;
+
+@mate.academy.lib.Dao
+public interface Dao<T> {
```
```suggestion
public interface BookDao {
```

liliia-ponomarenko commented on src/main/java/mate/academy/lib/dao/Dao.java:
```diff
@@ -0,0 +1,17 @@
+
+    T update(T t);
+
+    boolean deleteById(Long id);
```
remove parametrization

liliia-ponomarenko commented on src/main/java/mate/academy/lib/dao/impl/BookDao.java:
```diff
@@ -0,0 +1,113 @@
+import mate.academy.lib.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@mate.academy.lib.Dao
```
the same

liliia-ponomarenko commented on src/main/java/mate/academy/lib/dao/impl/BookDao.java:
```diff
@@ -0,0 +1,113 @@
+                Book book = new Book();
+                book.setId(resultSet.getLong("id"));
+                book.setTitle(resultSet.getString("title"));
+                book.setPrice(resultSet.getBigDecimal("price"));
```
can you move it to the private method and reuse?

liliia-ponomarenko commented on src/main/java/mate/academy/lib/dao/impl/BookDao.java:
```diff
@@ -0,0 +1,113 @@
+    
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new java.util.ArrayList<>();
```
use import

liliia-ponomarenko reviewed (changes requested):
Good job! A few comments ;)

svitlana-tertyshna committed:
Implemented BookDao

Checked comments: created a private method for repeated code; used necessary imports; removed parametrization

liliia-ponomarenko reviewed (approved):
Good job!
